apps/ios/Shared/Views/Chat/ChatView.swift

```diff
@@ -701,6 +714,80 @@ struct ChatView: View {
+                        if validItems.count > 0 {
+                            showAlert(
+                                String.localizedStringWithFormat(
+                                    NSLocalizedString("Forward %d message(s)?", comment: "alert title"),
+                                    validItems.count
+                                ),
+                                message: forwardConfirmation.reason + "\n" + NSLocalizedString(
+                                    "Forward messages without files?", 
+                                    comment: "alert message"
+                                )
+                            ) {
+                                switch forwardConfirmation {
+                                case let .filesNotAccepted(fileIds):
+                                    [forward(validItems), download(fileIds), alertAction("Cancel", style: .cancel)]
+                                default:
+                                    [forward(validItems), alertAction("Cancel", style: .cancel)]
+                                }
+                            }
+                        } else {
+                            showAlert(
+                                NSLocalizedString("Nothing to forward!", comment: "alert title"),
+                                message: forwardConfirmation.reason
+                            ) {
+                                switch forwardConfirmation {
+                                case let .filesNotAccepted(fileIds): 
+                                    [download(fileIds), alertAction("Cancel", style: .cancel)]
+                                default:
+                                    [alertAction("Ok")]
+                                }
+                            }
+                        }
+                    } else {
+                        openForwardingSheet(validItems)
+                    }
+                }
+            } catch {
+                logger.error("Plan forward chat items failed: \(error.localizedDescription)")
+            }
+        }
+
+        func forward(_ items: [Int64]) -> UIAlertAction {
+            alertAction(NSLocalizedString("Forward messages", comment: "alert action")) {
+                openForwardingSheet(items)
+            }
+        }
+
+        func download(_ fileIds: [Int64]) -> UIAlertAction {
+            alertAction(NSLocalizedString("Download files", comment: "alert action")) {
+                Task {
+                    if let user = ChatModel.shared.currentUser {
+                        await receiveFiles(user: user, fileIds: fileIds)
+                    }
+                }
+            }
+        }
+
+        func openForwardingSheet(_ items: [Int64]) {
+            let im = ItemsModel.shared
+            forwardedChatItems = items
+                .compactMap { id in im.reversedChatItems.first { $0.id == id } }
+                .sorted { $0.meta.itemTs == $1.meta.itemTs }
```

Comment:
It seems there is an issue with forwarding order in IOS, we do quite a bit of rounding to the second on the backend and it seems doing this sort isn't reliable. Can't we just keep the selection of the ids as before, and take advantage of the ordering already in the chat model and just filter here? That's the approach we took on android and it seems to be accurate